modules/@apostrophecms/asset/index.js

```diff
@@ -753,32 +757,28 @@ module.exports = {
               const jsFilename = JSON.stringify(component);
               const name = getComponentName(component, options, i);
               const jsName = JSON.stringify(name);
+              const importName = `${name}${options.importSuffix || ''}`;
               const importCode = `
-              import ${name}${options.importSuffix || ''} from ${jsFilename};
+              import ${importName} from ${jsFilename};
               `;
 
               output.paths.push(component);
               output.importCode += `${importCode}\n`;
 
               if (options.registerComponents) {
-                output.registerCode += `app.component(${jsName}, ${name});\n`;
+                output.registerCode += `window.apos.vueComponents[${jsName}] = ${importName};\n`;
```

Comment:
Where the suffix was not being taken into account. Matters when overrides are present.